iosApp/iosApp/ContentViewController.swift

```diff
@@ -11,33 +11,64 @@ import shared
 
 class ContentViewController: UIViewController {
 
+    @IBOutlet var tableView: UITableView!
+    
     let mainViewModel = MainViewModel()
     let kermit = Kermit(loggerList: [NSLogLogger()], defaultTag: "iOSTag")
 
+    var items: [CountryRow] = []
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        tableView.register(UINib(nibName: "CountryItemCell", bundle: nil), forCellReuseIdentifier: "CountryItemCell")
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        
         // Do any additional setup after loading the view.
         mainViewModel.setup(kermit: kermit)
-        mainViewModel.pageData.addObserver { data in
+        
+        mainViewModel.pageResultLD.addObserver { data in
```

Comment:
I don't know the exact way the memory graph is set up, but it seems you have a retain loop here. ContentViewController owns mainViewModel, which in turn holds this block, which strongly references self (aka the contentViewController). You can break this using `addObserver { [weak self] data in` here and/or in line 36. Also, I'd say that it's the MainViewModel's job to make the items available, so that line 19 isn't in this file.

Reply:
fixed. Is there a way to figure out if you have memory leaks in iOS app? We have a thing called Leak Canary in android

Reply:
<img width="338" alt="Screen Shot 2021-02-09 at 10 17 12" src="https://user-images.githubusercontent.com/68731/107342142-214ea780-6ac0-11eb-99fb-9104c4c81b8e.png">

The easiest way is to look at the memory graph and confirm you see the objects you'd expect to see: for example when you have navigated away from a view controller you'd expect it and all its views and view models to no longer show up.